shopinvader_utm/models/utm_mixin.py

```diff
@@ -0,0 +1,19 @@
+# Copyright 2021 ACSONE SA/NV (<http://acsone.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html)
+
+from odoo import api, models
+
+
+class UtmMixin(models.AbstractModel):
+    _inherit = "utm.mixin"
+
+    @api.model
```

Comment:
`@api.model` means `self` is empty (no recordset).
It's not correct because you do a `self.write(...)` just after.